packages/widgets/src/Pagination/Pagination.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { Button, SingleSelect, SingleSelectOption } from '@dhis2/ui'
+import { ChevronRight, ChevronLeft } from '@dhis2/ui-icons'
+
+import styles from './Pagination.styles.js'
+
+/**
+ * @module
+ * @param {Pagination.PropTypes} props
+ *
+ * @returns {React.Component}
+ *
+ * @example import { Pagination } from @dhis2/ui
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/pagination.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/pagination--default|Storybook}
+ */
+const Pagination = ({
+    dataTest,
+    hidePageSizeSelect,
+    hideGoToPageSelect,
```

Comment:
This feels like a little bit of an awkward API to me. Is it the intention to allow the dev to make the component more compact?

If so, I'd remove these in favor of a `compact` flag. That's inline with what some of our other components do, and seems to reflect the intention better. Also, it wouldn't feel as much like a double negative, which always makes it easier to understand at a glance.

Reply:
In general, I think we are adding these props mainly for the reason you suggested, to present users with a cleaner and more compact UI. In theory it could also be because a certain resource doesn't support a pageSize parameter... Probably this is only theoretical...

I agree the prop names are a bit awkward, but don't think you suggestion works because these are two elements and I think it needs to be possible to hide each one individually. Perhaps @cooper-joe can confirm.

If the above is true, the current prop names are the best I can come with, suggestions would be welcome 😄 ...

You mentioned something about double negatives, and I think this is a relative point to consider. First of all, this is not a double negative, even if you want to classify `hiding` as a negation then the words still only contain a single negative. But I do agree that having property names that instruct a component what it **should do** processes a lot better than ones that instruct the component what it **should not do**.....

In the past I have used positive words and `defaultProps` to sort of work around this. So then you'd get `showPageSizeSelect` with a default value of true, and the consumer of the component would have to set its value to `false` to hide that section. I suppose we could do that here too....

Reply:
Yeah I meant double in the way that hide... means _don't_ display, and the default is false. So the default is don't don't display componentX.

> In the past I have used positive words and defaultProps to sort of work around this. So then you'd get showPageSizeSelect with a default value of true, and the consumer of the component would have to set its value to false to hide that section. I suppose we could do that here too....

Yeah, that also seems like an awkward api. I wouldn't go in that direction either. If it's possible to rework the props to a positively worded one where the default is false, I'd prefer that. Otherwise I wouldn't know a better alternative either.